### DISCUSSION TITLE
Fail to create table with partition description with MAXVALUE

### DISCUSSION THREAD

**Discussion by xinhuitian** [apache/cloudberry#791] [Q&A] [answered]
Hello, I'm currently testing cb with the bootcamp 000-cbdb-sandbox docker file.
When I run a query similar as the below one:
```
gpadmin=# CREATE TABLE test_partition (
    name character varying,
    last_modified_date timestamp without time zone
)
WITH (appendoptimized=true, orientation=column, compresslevel=1)
DISTRIBUTED BY (name)
PARTITION BY RANGE (last_modified_date)
(
    PARTITION partition_202411 START ('2024-11-01 00:00:00') INCLUSIVE END ('2024-12-01 00:00:00') EXCLUSIVE,
    PARTITION partition_max START ('2024-12-01 00:00:00') INCLUSIVE END (MAXVALUE)
);
```
I got the following error:
```
ERROR:  cannot use column reference in partition bound expression
LINE 10: ...n_max START ('2024-12-01 00:00:00') INCLUSIVE END (MAXVALUE)
```

Split it to another query works fine:
```
CREATE TABLE partition_max partition of test_partition for values from ('2024-12-01 00:00:00') to (MAXVALUE);
```
Seems like MAXVALUE in the first query is parsed to a column ref. Is it as expected or a bug?

**Comment by avamingli** [comment]
Hi, @xinhuitian 

At first glance, it appears to be a bug. 
The syntax you're using seems to be specific to Greenplum rather than standard PostgreSQL grammar for defining a partitioned table and its partitions simultaneously. There may be some incompatibilities between the two.

If possible, could you create an issue for this instead of continuing the discussion? I appreciate you bringing this to our attention—thank you!

**Reply by xinhuitian** [reply]
Ok, I will create one

**Reply by tuhaihe** [reply]
Thanks!

Just one trick FYI: you can create one issue directly from the discussion.

<img width="276" alt="image" src="https://github.com/user-attachments/assets/b06f9c2e-43ea-424a-80ee-e8de998170ce" />

**Reply by xinhuitian** [reply, marked as answer]
https://github.com/apache/cloudberry/issues/795 Created

**Comment by avamingli** [comment]
Close as issue created.
